app/scripts/components/alerts.component.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .component('alerts', {
+    bindings: {
+      // Map (or array) of alert objects, each with properties
+      //   type:     'error', 'warning', 'success', or 'info' (default)
+      //   message:  the message to display
+      //   links:    an array of link objects (optional, see below)
+      //   onClose:  callback function when the user dismisses the alert (optional)
+      //
+      // Each link can have the following properties:
+      //   href:     link URL (optional if onClick is set)
+      //   label:    the link text
+      //   onClick:  handler to call when the link if clicked (optional)
+      //             If onClick returns true, dismisses the alert
+      alerts: '=',
+      // Filter function (optional).
+      filter: '=?',
+      animateSlide: '=?',
+      hideCloseButton: '=?',
+      toast: '=?'
+    },
+    templateUrl: 'views/_alerts.html',
+    controller: function() {
+      var ctrl = this;
+
+      ctrl.close = function(alert) {
+        alert.hidden = true;
+        if (_.isFunction(alert.onClose)) {
+          alert.onClose();
+        }
+      };
+
+      ctrl.onClick = function(alert, link) {
+        if (_.isFunction(link.onClick)) {
+          // If onClick() returns true, also hide the alert.
```

Comment:
Which `onClick` returns `true`?  `ctrl.onClick` or `links.onClick`?  This comment is a bit confusing.

Reply:
(preference for no comment if the code can be altered a bit to make that clear)

Reply:
Most of the comments were just copy/paste from directive, but I can go back and spruce them up if needed.

Reply:
Can you update the comment to say `link.onClick()`?

I think the comment here is helpful.